internal/resources/aws/instance.go

```diff
@@ -312,3 +271,41 @@ func (a *Instance) cpuCreditCostComponent(instanceFamily string) *schema.CostCom
 		},
 	}
 }
+
+// EC2 implementation of reservationResolver
+type ec2ReservationResolver struct {
+	term              string
+	paymentOption     string
+	termOfferingClass string
+}
+
+func newEc2ReservationResolver(term, paymentOption, termOfferingClass string) reservationResolver {
+	return &ec2ReservationResolver{
+		term:              term,
+		paymentOption:     paymentOption,
+		termOfferingClass: termOfferingClass,
+	}
+}
```

Comment:
See other comments about godoc and constructors

Reply:
Removed constructor